Manager/PaymentManager.php

```diff
@@ -32,39 +33,92 @@ class PaymentManager
+    public function getAllPaymentGatewayConfigurationFromDoctrine(): array
     {
-        return $this
+        $paymentGatewayConfigurations = $this
             ->om
             ->getRepository(PaymentGatewayConfiguration::class)
             ->findAll()
         ;
+
+        foreach ($paymentGatewayConfigurations as $key => $paymentGatewayConfiguration) {
+            $paymentGatewayConfigurations[$paymentGatewayConfiguration->getAlias()] = $paymentGatewayConfiguration;
+            unset($paymentGatewayConfigurations[$key]);
+        }
+
+        return $paymentGatewayConfigurations;
+    }
+
+    public function getAllPaymentGatewayConfigurationFromConfiguration(): array
+    {
+        $paymentGatewayConfigurations = [];
+
+        foreach ($this->gatewayConfigurations as $alias => $configuration) {
+            if (!isset($configuration['gateway_name'])) {
+                throw new InvalidPaymentGatewayConfigurationException(
+                    sprintf('No gateway name given for the payment gateway configuration %s', $alias)
+                );
+            }
+
+            $paymentGatewayList = $this->paymentGatewayRegistry->getAll();
+            $paymentGatewayFQCN = get_class($paymentGatewayList[$configuration['gateway_name']]);
+
+            foreach ($paymentGatewayFQCN::getParameterNames() as $parameterName) {
+                if (!array_key_exists($parameterName, $configuration['parameters'])) {
+                    throw new InvalidPaymentGatewayConfigurationException(
+                        sprintf('Parameter %s not found for payment gateway configuration %s', $parameterName, $alias)
+                    );
+                }
+            }
+
+            $paymentGatewayConfigurations[$alias] = (new PaymentGatewayConfiguration())
+                ->setAlias($alias)
+                ->setGatewayName($configuration['gateway_name'])
+                ->setEnabled($configuration['enabled'])
+                ->setParameters($configuration['parameters'])
+            ;
+        }
+
+        return $paymentGatewayConfigurations;
+    }
+
+    public function getAllPaymentGatewayConfiguration(): array
+    {
+        return array_merge(
+            $this->getAllPaymentGatewayConfigurationFromDoctrine(),
+            $this->getAllPaymentGatewayConfigurationFromConfiguration()
+        );
     }
 
     public function createPaymentContextByAlias(string $alias): PaymentContextInterface
     {
-        $paymentGatewayConfiguration = $this
-            ->om
-            ->getRepository(PaymentGatewayConfiguration::class)
-            ->findOneBy(['alias' => $alias])
-        ;
+        $paymentGatewayConfigurations = $this->getAllPaymentGatewayConfiguration();
 
-        if (null === $paymentGatewayConfiguration) {
+        if (!isset($paymentGatewayConfigurations[$alias])) {
             throw new NoPaymentGatewayConfigurationFoundException();
         }
 
+        $paymentGatewayConfiguration = $paymentGatewayConfigurations[$alias];
```

Comment:
replace all this previous code with a single function that do "getPaymentGatewayConfiguration($alias)"